kore/src/Kore/Strategies/Goal.hs

```diff
@@ -1011,17 +1001,23 @@ removalPredicate
     Conditional { term = destTerm } = destination
     (configTerm, configPredicate) = Pattern.splitTerm configuration
     sideCondition = SideCondition.assumeTrueCondition configPredicate
+    configSort = termLikeSort configTerm
+    remainderPattern patt =
+        Pattern.fromCondition
+        . Conditional.withoutTerm
+        $ (const <$> destination <*> patt)
```

Comment:
I know you didn't originally write this part, but you might consider: 

```suggestion
        Pattern.fromCondition
        $ on (<>) Conditional.withoutTerm destination patt
```

I think it's not obvious that `(const <$> _ <*> _)` is the `Semigroup` action on the conditions.